src/index.ts

```diff
@@ -168,6 +170,9 @@ export const htmlPlugin = (configuration: Configuration = { files: [], }): esbui
                 const htmlFileDirectory = posixJoin(outDir, htmlFileConfiguration.filename)
                 targetPath = path.relative(path.dirname(htmlFileDirectory), filepath)
             }
+            if (htmlFileConfiguration.hash) {
+                targetPath = `${targetPath}?${crypto.createHash('md5').update(JSON.stringify(outputFile)).digest("hex")}`;
```

Comment:
I'm not sure if it's enough to base the cache-busting effect on the metadata. While there is probably not a high probability for this, if you just change a typo in one of your files and the number of characters stays the same, the metadata won't change, and the hash would not change :/

Reply:
yeah, I thought about that but thought this might be close enough. If we don't care for reproducible builds, we could salt with the build time as well (or just do the build timestamp instead of the metafile info).

Reply:
Updated this to either hash the current time, or to allow folks to pass through a string for the base of the hash (such as the git commit SHA).